espnet/nets/pytorch_backend/transformer/embedding.py

```diff
@@ -1,54 +1,90 @@
+"""Positonal Encoding Module."""
 import math
 
 import torch
 
 
 class PositionalEncoding(torch.nn.Module):
-    """Positional encoding module
-
-    :param int d_model: embedding dim
-    :param float dropout_rate: dropout rate
-    :param int max_len: maximum input length
-    """
+    """Positional encoding."""
 
     def __init__(self, d_model, dropout_rate, max_len=5000):
+        """Initialize class.
+
+        :param int d_model: embedding dim
+        :param float dropout_rate: dropout rate
+        :param int max_len: maximum input length
+
+        """
         super(PositionalEncoding, self).__init__()
+        self.d_model = d_model
+        self.xscale = math.sqrt(self.d_model)
         self.dropout = torch.nn.Dropout(p=dropout_rate)
-        # Compute the positional encodings once in log space.
-        pe = torch.zeros(max_len, d_model)
-        position = torch.arange(0, max_len, dtype=torch.float32).unsqueeze(1)
-        div_term = torch.exp(torch.arange(0, d_model, 2, dtype=torch.float32) *
-                             -(math.log(10000.0) / d_model))
+        self.pe = None
+        self.reset_(torch.tensor(0.0).expand(1, max_len))
+
+    def reset_(self, x):
```

Comment:
`extend_positional_encoding` or `extend_pe` may be better than `reset` as the function name since `reset(x)` may not be understandable.